### PR TITLE
Linspace accepts sympy symbol

### PR DESCRIPTION
**Pull request by timothyman**
- Added K as type for key which can be `str` or `sympy.Symbol`
- Added call to `str` in `SingleSweep` class constructor

Fixes https://github.com/quantumlib/Cirq/issues/1397

### CODE REVIEW

**Comment by timothyman** [comment]
I signed it!

**Comment by timothyman** [comment]
I signed it!

File: cirq/study/sweeps.py [hunk 1]
```diff
@@ -240,11 +240,14 @@ def __str__(self):
...
 
+K = TypeVar('K', str, Symbol)  # Key can be of type str or sympy.Symbol
+
+
 class SingleSweep(Sweep):
     """A simple sweep over one parameter with values from an iterator."""
 
-    def __init__(self, key: str) -> None:
-        self.key = key
+    def __init__(self, key: K) -> None:
```
Comment (Strilanc) [inline cirq/study/sweeps.py]: Nit: I would just inline `Union[str, sympy.Symbol]` in each case.

File: cirq/study/sweeps.py [hunk 2]
```diff
@@ -240,11 +240,14 @@ def __str__(self):
         return ' + '.join(str(s) for s in self.sweeps)
 
 
+K = TypeVar('K', str, Symbol)  # Key can be of type str or sympy.Symbol
```
Comment (Strilanc) [inline cirq/study/sweeps.py]: You want `Union[str, sympy.Symbol]` instead of a type variable.

**Comment by timothyman** [comment]
- Use `import sympy` instead of `from sympy import Symbol` to make `sympy.Symbol` explicit
- Replaced `TypeVar` for `Union[str, sympy.Symbol]`
- Added type-check for `sympy.Symbol` before calling `str` in `SingleSweep` class constructor

**Pull request merged**